File: src/src/classes/ContactPerson.ts
import {Patient} from "./Patient.ts";
import {Salts} from "./Salt.ts";
import {Hashed} from "./Hashed.ts";

export class ContactPerson{
    id: number;
    firstName: string;
    lastName: string;
    phone: string;
    relationship: string;
    patient: Patient;
    salts: Salts;


    constructor(id: number, firstName: string, lastName: string, phone: string,
                relationship: string, patient: Patient, salts: Salts) {
        this.id = id;
        this.salts = salts;

        this.firstName = Hashed.extractDataFromHash(firstName,this.salts.firstName);
        this.lastName = Hashed.extractDataFromHash(lastName,this.salts.lastName);
        this.phone = Hashed.extractDataFromHash(phone,this.salts.phone);
        this.relationship = Hashed.extractDataFromHash(relationship,this.salts.relationship);


        this.patient = new Patient(patient.id,patient.PNC,patient.firstName,
            patient.lastName,patient.mail, patient.phone, patient.psychiatrist,
            patient.psychotherapist,patient.salts
        );

    }
}

File: src/main/java/com/example/airmed/Entity/ContactPerson.java
package com.example.airmed.Entity;
import com.example.airmed.Hashed;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.HashMap;
import java.util.Map;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class ContactPerson {
    @Id
    @GeneratedValue(strategy =  GenerationType.IDENTITY)
    @Column(name = "id", nullable = false, updatable = false)
    private Long id;

    @Column(nullable = false)
    private String firstName;

    @Column(nullable = false)
    private String lastName;

    @Column(nullable = false)
    private String phone;

    @Column(nullable = false)
    private String relationship;

    @ManyToOne
    @JoinColumn(name = "patient", nullable = false)
    private Patient patient;

//    @ElementCollection
//    private Map<String,String> salts = new HashMap<>();
//
//    public void setFirstName(String firstName) {
//        String salt = Hashed.generateSalt();
//        this.salts.put("firstName", salt);
//        this.firstName = Hashed.createHashData(firstName,salt);
//    }
//
//    public void setLastName(String lastName) {
//        String salt = Hashed.generateSalt();
//        this.salts.put("lastName", salt);
//        this.lastName = Hashed.createHashData(lastName,salt);
//    }
//
//    public void setPhone(String phone) {
//        String salt = Hashed.generateSalt();
//        this.salts.put("phone", salt);
//        this.phone = Hashed.createHashData(phone,salt);
//    }
//
//    public void setRelationship(String relationship) {
//        String salt = Hashed.generateSalt();
//        this.salts.put("relationship", salt);
//        this.relationship = Hashed.createHashData(relationship,salt);
//    }
}
